src/sonic_ax_impl/mibs/__init__.py

```diff
@@ -148,10 +150,58 @@ def init_db():
     """
     # SyncD database connector. THIS MUST BE INITIALIZED ON A PER-THREAD BASIS.
     # Redis PubSub objects (such as those within swsssdk) are NOT thread-safe.
-    db_conn = SonicV2Connector(**redis_kwargs)
+    db_conn = SonicV2Connector(**redis_kwargs) 
 
     return db_conn
 
+def init_multi_db():
+    db_conn= []
+    if (os.path.isfile('/usr/share/sonic/platform/asic.conf')):
+        fp = open('/usr/share/sonic/platform/asic.conf', 'r')
+        line = fp.readline()
+        while line:
+            line_split = line.split('=')
+            if line_split[0].lower() == "num_asic":
+                num_asic=int(line_split[1])
+                break
+            line = fp.readline()
+        fp.close()
+    else:
+        num_asic = 1 
+
+    if num_asic == 1:
+        db = SonicV2Connector(use_unix_socket_path=True)
+        db_conn.append(db)
+        return db_conn
+
+    for asic_inst in range(num_asic):
+        namespace = "asic" + str(asic_inst)
+        db = SonicV2Connector(use_unix_socket_path=True, namespace=namespace)
+        db_conn.append(db)
+
+    return db_conn
+
+def get_keys_from_multi_db(multi_db_conn, db_id, pattern='*'):
+    result_keys=[]
+    for db_conn in multi_db_conn:
+        db_conn.connect(db_id)
+        keys = db_conn.keys(db_id, pattern)
+        result_keys.append(keys)
+    return result_keys
+
+
+def get_all_from_multi_db(global_db_conn, multi_db_conn, db_name, _hash, *args, **kwargs):
+    # check if key exisits in global db
+    global_db_conn.connect(db_name)
+    if (global_db_conn.exists(db_name, _hash)):
+        return global_db_conn.get_all(db_name, _hash, *args, **kwargs)
+   
+    for db_conn in multi_db_conn:
+        db_conn.connect(db_id)
+        if(db_conn.exists(db_name, _hash)):
+            return db_conn.get_all(db_name, _hash, *args, **kwargs)
+    return {}
```

Comment:
They are common functions, can we move it to a common place so others can reuse?

Reply:
Does not look like any other application will require this, also other application will need specific processing. Will leave this as is.

Reply:
How about reuse by pmon/cli? If yes, move common functions or new class to sonic-py-swsssdk.

---
In reply to: [416238749](https://github.com/Azure/sonic-snmpagent/pull/126#discussion_r416238749) [](ancestors = 416238749)

Reply:
based on current design changes being other components do not require these functions. Will revisit or modify if it is required in future.